themes/digital.gov/src/js/github.js

```diff
@@ -1,89 +1,158 @@
-  });
-}
-
-jQuery(($) => {
-  function buildEditFileLink() {
-    // editpathURL is set the <head>
-    if (editpathURL !== undefined) {
-      // Build the edit link
-      const edit = `<a target='_blank' class='edit-file-link' href='${editpathURL}' title='Edit in GitHub'>Edit</a>`;
-
-      // Insert the .edit-file-link html into the .edit-file div and remove the .hidden class
-      $("#page-data .edit-file").html(edit).removeClass("hidden");
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
+  const editFileButton = document.querySelector(".edit-file");
+  const editPageLink = document.querySelector("#page-data .edit-file");
+  const gitRepo = {};
+
+  /**
+   * Get hugo file path from the data-edit-this attribute to link to Github repo location
+   * example string: news/2023/07/2023-07-19-gsa-shared-service-provider-program-guide.md
+   */
+  function getFilepath() {
+    if (editPageLink) {
+      gitRepo.filepath = document
+        .querySelector("div[data-edit-this]")
+        .getAttribute("data-edit-this");
+    } else {
+      gitRepo.filepath = "";
+    }
+  }
+
+  /**
+   * Set the branch from the URL path
+   * If on cloud.pages get the branch name from the URL
+   * Otherwise, use main for localhost and production
+   */
+  function getBranch() {
+    const host = window.location.hostname;
+    if (host.includes("sites.pages.cloud.gov")) {
+      // eslint-disable-next-line prefer-destructuring
+      gitRepo.branch = window.location.pathname.split("/")[4];
+    } else {
+      gitRepo.branch = "main";
+    }
+  }
+
+  /**
+   * format github ISO date format to human friendly datetime string
+   * @param {timezone} timezoneDate YYYY-MM-DDTHH:MM:SSZ
+   * @returns {string} Formatted string that reads "Jul 6, 2023 at 5:23 p.m., ET"
+   */
+  function formatDate(timezoneDate) {
+    const inputDate = new Date(timezoneDate);
+
+    const dateOptions = {
+      day: "numeric",
+      month: "short",
+      year: "numeric",
+    };
+
+    const timeOptions = {
+      hour: "numeric",
+      minute: "2-digit",
+      timeZone: "America/New_York",
+      timeZoneName: "shortGeneric",
+    };
+
+    const outputDate = inputDate.toLocaleDateString(undefined, dateOptions);
+    const outputTime = inputDate
+      .toLocaleTimeString("en-US", timeOptions)
+      .replace("AM", "a.m.,")
+      .replace("PM", "p.m.,");
+    return `${outputDate} at ${outputTime}`;
+  }
+
+  /**
+   * Add "Edit" link to the #page-data element before the "Last updated on" commit timestamp
+   * Checks if page has the #page-data element for single pages, not displayed on landing pages
+   */
+  function displayGithubEditLink() {
+    const gitEditFilePath = `https://github.com/GSA/digitalgov.gov/edit/${gitRepo.branch}/content/${gitRepo.filepath}`;
+    if (gitEditFilePath !== undefined) {
```

Comment:
If this is empty or `null` by default we can simplify this check.

```suggestion
    if (gitEditFilePath) {
```